Build and test on ARM Android.

pietro opened the pull request.

briansmith commented on .travis.yml:
```diff
@@ -94,6 +96,30 @@ matrix:
+            - openjdk-6-jre-headless
+
+    - env: TARGET_X=arm-linux-androideabi CC_X=arm-linux-androideabi-gcc CXX_X=arm-linux-androideabi-g++ MODE_X=DEBUG KCOV=0
+      rust: nightly
```
You are missing the `rust: nightly` + `MODE_X=RELWITHDEBINFO` option. Please add it.

pietro replied:
Added.

briansmith commented on mk/travis.sh:
```diff
@@ -61,7 +69,27 @@ rustc --version
+  adb shell mkdir -p /data/third-party/NIST
+  adb push crypto/poly1305/poly1305_test.txt /data/crypto/poly1305/poly1305_test.txt
+  adb push crypto/cipher/test /data/crypto/cipher/test
+  for testfile in `ls src/*_tests.txt`; do adb push $testfile /data/$testfile ; done
```
Can't we do something like this?:
```
for testfile in `find src crypto -name "*_tests.txt"`; do
   adb shell mkdir -p `dirname $testfile`
   ....
````
?

briansmith commented:
Please add the "I agree..." statement to the commit message (wrapped at 72 characters) and address the two nits.

This looks awesome!

pietro commented:
I added the wrapped  "I agree..." statement to the commit message. I didn't add it as a comment to the PR though.

briansmith commented:
Unfortunately, the build fails:
```
+adb wait-for-device
+emulator64-arm @arm-18 -no-audio -no-window
emulator64-arm: error while loading shared libraries: libQt5Widgets.so.5: cannot open shared object file: No such file or directory
* daemon not running. starting it now on port 5037 *
* daemon started successfully *
```

pietro commented:
Weird I haven't had that issue both locally and on travis. I'll do some more testing.

pietro commented:
Sorry. I fixed the build and added a few things:

1. `travis_wait` to the android builds because the long running tests on release builds.
2. logic to copy the right binary (debug or release) to the emulator image.

@briansmith can you please review those? thanks.

briansmith commented:
Awesome! Thanks! Landed as 72616bbf0580a059b844ffb42e6d8830073df56d.
